refactored query stats: implemetation instead interface (#1610)
ydb-platform/ydb-go-sdk: CHANGELOG.md

> * Added into query stats the immutable range iterators from go1.23 for iterate over query phases and accessed tables   

Author: ```suggestion
* Added immutable range iterators from go1.23 into query stats to iterate over query phases and accessed tables without query stats object mutation
```
